src/R/Interpreters/Impl/Implementation/RInterpreterInfo.cs

```diff
@@ -63,36 +63,28 @@ public sealed class RInterpreterInfo : IRInterpreterInfo {
                     fs.FileExists(rGuiPath)) {
 
                     var fileVersion = GetRVersionFromBinary(fs, rDllPath);
-                    if (fileVersion == Version && (svr == null || svr.IsCompatibleVersion(Version))) {
-                        _isValid = true;
+                    _isValid = fileVersion == Version && svr.IsCompatibleVersion(Version);
+                    if (!_isValid.Value) {
+                        coreShell?.ShowMessage(
+                            string.Format(CultureInfo.InvariantCulture, Resources.Error_UnsupportedRVersion,
+                            Version.Major, Version.Minor, Version.Build, svr.MinMajorVersion, svr.MinMinorVersion, "*",
+                            svr.MaxMajorVersion, svr.MaxMinorVersion, "*"), MessageButtons.OK);
                     }
-
-                    coreShell?.ShowMessage(
-                        string.Format(CultureInfo.InvariantCulture, Resources.Error_UnsupportedRVersion,
-                        Version.Major, Version.Minor, Version.Build, svr.MinMajorVersion, svr.MinMinorVersion, "*",
-                        svr.MaxMajorVersion, svr.MaxMinorVersion, "*"), MessageButtons.OK);
-
                 } else {
                     coreShell?.ShowMessage(string.Format(CultureInfo.InvariantCulture, Resources.Error_CannotFindRBinariesFormat, InstallPath), MessageButtons.OK);
                 }
-            } catch(IOException ioex) {
-                ex = ioex;
-            } catch (ArgumentException aex) {
-                ex = aex;
-
-            } catch(UnauthorizedAccessException uaex) {
-                ex = uaex;
-            }
+            } catch (IOException ioex) { ex = ioex; } catch (ArgumentException aex) { ex = aex; } catch (UnauthorizedAccessException uaex) { ex = uaex; }
```

Comment:
maybe it will look better:
```{csharp}
} catch (Exception ex) when (ex is IOException || ex is ArgumentException || ex is UnauthorizedAccessException)
    coreShell?.ShowErrorMessage(
                    string.Format(CultureInfo.InvariantCulture, Resources.Error_ExceptionAccessingPath, InstallPath, ex.Message));
}
```
Up to you